napari/utils/colormaps/colormap.py

```diff
@@ -177,43 +192,25 @@ def map(self, values) -> np.ndarray:
         """
         values = np.atleast_1d(values)
 
-        precast = _cast_labels_to_minimum_dtype_auto(
-            values, len(self.colors) - 1, self.background_value
-        )
-
-        return self._map_precast(precast)
-
-    def _map_precast(self, values) -> np.ndarray:
-        """Map *precast* values to colors.
-
-        When mapping values, we first convert them to a smaller dtype for
-        performance reasons. This conversion changes the label values,
-        even for small labels. This method is used to map values that have
-        already been converted to the smaller dtype.
+        if values.dtype.kind == 'f':
+            values = values.astype(np.int64)
 
-        Parameters
-        ----------
-        values : np.ndarray
-            Values to be mapped. They must have already been downcast using
-            `_cast_labels_to_minimum_dtype_auto`.
-
-        Returns
-        -------
-        np.ndarray of shape (N, M, 4)
-            Mapped colors.
-        """
-        mapped = self.colors[values.astype(np.int64)]
-
-        mapped[values == self.background_value] = 0
-
-        # If using selected, disable all others
-        if self.use_selection:
-            cast_selection = _cast_labels_to_minimum_dtype_auto(
-                np.array([self.selection]),
-                len(self.colors) - 1,
-                self.background_value,
-            )[0]
-            mapped[values != cast_selection] = 0
+        if values.dtype == np.uint8 and "_uint8_colors" in self.__dict__:
+            mapped = self._uint8_colors[values]
+        elif values.dtype == np.uint16 and "_uint16_colors" in self.__dict__:
+            mapped = self._uint16_colors[values]
+        else:
+            background = int(
+                np.array([self.background_value]).astype(values.dtype)[0]
```

Comment:
as above could be:

```suggestion
                np.array(self.background_value).astype(values.dtype)[0]
```

Reply:
No. Because above we use `values = np.atleast_1d(values)` so we could pass single number. Here it will not produce valid array.

Reply:
oops, somehow I thought this was a `map` as well :/